pkg/apiserver/rest/apis/v1/types.go

```diff
@@ -20,6 +20,58 @@ import (
 	"time"
 )
 
+// CreateAddonRequest defines the format for addon create request
+type CreateAddonRequest struct {
+	Name string `json:"name" validate:"required"`
+
+	Description string `json:"description,omitempty"`
+
+	Icon string `json:"icon"`
+
+	Initializer string `json:"initializer"`
+}
+
+// ListAddonResponse defines the format for addon list response
+type ListAddonResponse struct {
+	Addons []AddonMeta `json:"addons"`
+}
+
+// AddonStatusResponse defines the format of addon status response
+type AddonStatusResponse struct {
+}
+
+// AddonPhase defines the phase of an addon
+type AddonPhase string
+
+const (
+	// AddonPhaseDisabled indicates the addon is disabled
+	AddonPhaseDisabled AddonPhase = "disabled"
+	// AddonPhaseDisabling indicates the addon is disabling
+	AddonPhaseDisabling AddonPhase = "disabling"
+	// AddonPhaseEnabled indicates the addon is enabled
+	AddonPhaseEnabled AddonPhase = "enabled"
+	// AddonPhaseEnabling indicates the addon is enabling
+	AddonPhaseEnabling AddonPhase = "enabling"
+)
+
+// AddonMeta defines the format for a single addon
+type AddonMeta struct {
+	Name string `json:"name"`
+
+	Description string `json:"description"`
+
+	Icon string `json:"icon"`
+
+	Phase AddonPhase `json:"phase"`
+}
+
+// DetailAddonResponse defines the format for showing the addon details
+type DetailAddonResponse struct {
+	AddonMeta
+
```

Comment:
More addon information should be enriched, such as which XDefinition. Corresponding applications, etc.

Reply:
Adding.